### PR TITLE
SignalR (Server & Client) fundamentals

### PR DESCRIPTION
**Pull request by adambrodin**
### **Main changes:**
- Split games up into groups
- Basic turn management
- Basic dice rolling
- Bringing out token basics
- Changed protocol to MessagePack for extra low latency & smaller message sizes
- Join/leave message + user removal from game on disconnect

_**Please do not remove branch after merge!**_

### CODE REVIEW

File: src/LinuxLudo.API/Domain/Models/ConnectedUser.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+using LinuxLudo.Core.Models;
+
+namespace LinuxLudo.API.Domain.Models
+{
+    public class ConnectedUser
+    {
+        public string Username { get; set; }
+        public string ConnectionId { get; set; }
+        public OpenGame JoinedGame { get; set; }
```
Comment (TheNerdyHamster) [inline src/LinuxLudo.API/Domain/Models/ConnectedUser.cs]: We might wanna have private set on these due to they shouldnt be possible to modify after created?

File: src/LinuxLudo.API/Hubs/GameHub.cs [hunk 2]
```diff
@@ -25,19 +24,74 @@ public async Task JoinGame(string username, Guid gameId)
...
+
+                // Removes the player from their game
+                _repository.RemovePlayer(user.JoinedGame, user.Username);
+
+                // Update the clients that user has left
+                await SendConnectionChanged(user.JoinedGame.GameId.ToString(), user.Username, _repository.FetchGameById(user.JoinedGame.GameId).PlayersInGame);
+
+            }
+
+            await base.OnDisconnectedAsync(exception);
```
Comment (TheNerdyHamster) [inline src/LinuxLudo.API/Hubs/GameHub.cs:189]: To make it abit easier to read and understand try to switch the if stmt around
```suggestion
            if (user?.JoinedGame == null)
            {
                await base.OnDisconnectedAsync(exception);
                return
            }

            if (user.JoinedGame.CurrentTurnColor == user.JoinedGame.PlayersInGame.First(player => player.Name == user.Username).Color)
            {
               await UpdatePlayerTurn(user.JoinedGame);
            }

            _repository.RemovePlayer(user.JoinedGame, user.Username);

            await SendConnectionChanged(user.JoinedGame.GameId.ToString(), user.Username, _repository.FetchGameById(user.JoinedGame.GameId).PlayersInGame);
            await base.OnDisconnectedAsync(exception);
```

**Review by TheNerdyHamster — DISMISSED** [review]
Everything looks good!

File: src/LinuxLudo.API/Domain/Models/ConnectedUser.cs (hunk 1 above)
Comment (adambrodin) [reply src/LinuxLudo.API/Domain/Models/ConnectedUser.cs]: They wont be accessible as privates.

File: src/LinuxLudo.API/Hubs/GameHub.cs (hunk 2 above)
Comment (adambrodin) [reply src/LinuxLudo.API/Hubs/GameHub.cs:189]: Repository RemovePlayer method cannot be called if user.Joined game is null. - Leaving as is for now.

**Review by TheNerdyHamster — APPROVED** [review]

**Pull request merged**